src/main/java/se/liu/ida/hefquin/engine/queryplan/utils/TextBasedPhysicalPlanPrinterImpl2.java

```diff
@@ -0,0 +1,354 @@
+	}
+	
+	protected void printOperatorInfoForRequest( final PhysicalOpRequest op, final PrintStream out, final String indentLevelString, final String indentLevelStringForOpDetail ) {
+		final LogicalOpRequest lop = op.getLogicalOperator();
+		final DataRetrievalRequest req = lop.getRequest();
+		out.append( indentLevelString + "req (" + lop.getID() + ")" );
+		out.append( System.lineSeparator() );
+		printFederationMember( lop.getFederationMember(), indentLevelStringForOpDetail, out );
+		out.append( indentLevelStringForOpDetail + "  - pattern (" + req.hashCode() +  ") (" + req.toString() + ")" );
+		out.append( System.lineSeparator() );
+		out.append( indentLevelStringForOpDetail );
+		out.append( System.lineSeparator() );
+	}
+	
+	protected void printOperatorInfoForOpSymmetricHashJoin ( final PhysicalOpSymmetricHashJoin op, final PrintStream out, final String indentLevelString, final String indentLevelStringForOpDetail ) {
+		out.append( indentLevelString + "symmetricHashJoin (" + op.getID() + ") " );
+		printLogicalOpForPhysicalOp( op.getLogicalOperator(), out, indentLevelString, indentLevelStringForOpDetail );
+	}
+	
+	protected void printLogicalOpForPhysicalOp ( final LogicalOperator lop, final PrintStream out, final String indentLevelString, final String indentLevelStringForOpDetail ) {
+		out.append( System.lineSeparator() );
+		if ( lop instanceof LogicalOpUnion ) {
+			printLogicalOpUnionForPhysicalOp( (LogicalOpUnion) lop, out, indentLevelStringForOpDetail );
+		}
+		else if ( lop instanceof LogicalOpMultiwayUnion ) {
+			printLogicalOpMultiwayUnionForPhysicalOp( (LogicalOpMultiwayUnion) lop, out, indentLevelStringForOpDetail );
+		}
+		else if ( lop instanceof LogicalOpTPAdd ) {
+			printLogicalOpTpAddForPhysicalOp( (LogicalOpTPAdd) lop, out, indentLevelStringForOpDetail );
+		}
+		else if ( lop instanceof LogicalOpTPOptAdd ) {
+			printLogicalOpTpOptAddForPhysicalOp( (LogicalOpTPOptAdd) lop, out, indentLevelStringForOpDetail );
+		}
+		else if ( lop instanceof LogicalOpBGPAdd ) {
+			printLogicalOpBGPAddForPhysicalOp( (LogicalOpBGPAdd) lop, out, indentLevelStringForOpDetail );
+		}
+		else if ( lop instanceof LogicalOpBGPOptAdd ) {
+			printLogicalOpBGPOptAddForPhysicalOp( (LogicalOpBGPOptAdd) lop, out, indentLevelStringForOpDetail );
+		}
+		else if ( lop instanceof LogicalOpGPAdd ) {
+			printLogicalOpGPAddForPhysicalOp( (LogicalOpGPAdd) lop, out, indentLevelStringForOpDetail );
+		}
+		else if ( lop instanceof LogicalOpGPOptAdd) {
+			printLogicalOpGPOptAddForPhysicalOp( (LogicalOpGPOptAdd) lop, out, indentLevelStringForOpDetail );
+		}
+		else if ( lop instanceof LogicalOpFilter ) {
+			printLogicalOpFilterForPhysicalOp( (LogicalOpFilter) lop, out, indentLevelStringForOpDetail );
+		}
+		else if ( lop instanceof LogicalOpGlobalToLocal ) {
+			printLogicalOpGlobalToLocalForPhysicalOp( (LogicalOpGlobalToLocal) lop, out, indentLevelStringForOpDetail );
+		}
```

Comment:
All of this stuff with all these additional `printLogicalOp...ForPhysicalOp` functions is too complicated. Why not simply a function that returns the short string name for each of the logical operators (e.g., `"tpAdd"` for `LogicalOpTPAdd`) and then a generic function that puts that short string name together with the rest of the things needed for the `- lop` line (where this rest is only the string `- lop` and the string with the ID of the log.op. in parenthesis).

Reply:
I'll update as you suggest.